src/main/java/org/jvnet/hudson/plugins/m2release/ForkedRepoVersionInfo.java

```diff
@@ -0,0 +1,99 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+package org.jvnet.hudson.plugins.m2release;
+
+import org.apache.maven.shared.release.versions.DefaultVersionInfo;
+import org.apache.maven.shared.release.versions.VersionInfo;
+import org.apache.maven.shared.release.versions.VersionParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ForkedRepoVersionInfo extends DefaultVersionInfo {
+
+    private transient Logger log = LoggerFactory.getLogger(ForkedRepoVersionInfo.class);
+
+    public final static Pattern FORKED_REPO_NEXT_DEVELOPMENT_VERSION_PATTERN = Pattern.compile("(.*)(((-wso2v)(([1-9])(\\d*)))(-SNAPSHOT))$");
+    private final static int MAJOR_MINOR_PATCH_VERSION_GROUP_NUM = 1;
```

Comment:
This constant's purpose is a bit confusing. Shall we improve the constant name?

Reply:
how about "DEFAULT_VERSION_GROUP_INDEX

Reply:
+1.. `DEFAULT_VERSION_REGEX_GROUP_INDEX`